src/cargo/util/toml/mod.rs

```diff
@@ -368,6 +368,42 @@ pub fn prepare_for_publish(
         lints: me.lints.clone(),
     });
 
+    fn map_feature(
+        features: Option<&BTreeMap<manifest::FeatureName, Vec<String>>>,
+        deps: Option<&BTreeMap<manifest::PackageName, manifest::InheritableDependency>>,
+    ) -> CargoResult<Option<BTreeMap<manifest::FeatureName, Vec<String>>>> {
+        let Some(features) = features else {
+            return Ok(None);
+        };
+        let features = features
+            .iter()
+            .map(|(name, feature_deps)| {
+                let feature_deps = feature_deps
+                    .iter()
+                    .filter(|feature_dep| {
+                        if let Some(feature_pkg_name) = feature_dep.split('/').next() {
```

Comment:
We might want to use [`FeatureValue`](https://doc.rust-lang.org/nightly/nightly-rustc/cargo/core/summary/enum.FeatureValue.html) instead of parsing it again.